src/clients/python/main.py

```diff
@@ -0,0 +1,389 @@
+from pathlib import Path
+import aiofiles
+import httpx
+from pyee.asyncio import AsyncIOEventEmitter
+import msgspec
+from uuid import uuid4
+import sys
+
+# Import schemas
+from schemas.WebViewMessage import (
+    NotificationMessage,
+    ResponseMessage,
+    StartedNotification,
+    ClosedNotification,
+    StringResultType,
+    BooleanResultType,
+    SizeResultType,
+    ResultType,
+    AckResponse,
+    ResultResponse,
+    ErrResponse,
+    Notification as WebViewNotification,
+    WebViewMessage,
+)
+from schemas.WebViewRequest import (
+    WebViewRequest,
+    GetVersionRequest,
+    EvalRequest,
+    SetTitleRequest,
+    GetTitleRequest,
+    SetVisibilityRequest,
+    IsVisibleRequest,
+    OpenDevToolsRequest,
+    GetSizeRequest,
+    SetSizeRequest,
+    FullscreenRequest,
+    MaximizeRequest,
+    MinimizeRequest,
+    LoadHtmlRequest,
+    LoadUrlRequest,
+    SimpleSize,
+)
+from schemas.WebViewOptions import (
+    WebViewOptions,
+    WebViewContentHtml,  # type: ignore  # noqa: F401
+)
+
+# Constants
+BIN_VERSION = "0.1.14"
+
+T = TypeVar("T", bound=WebViewNotification)
+
+
+def return_result(
+    result: Union[AckResponse, ResultResponse, ErrResponse],
+    expected_type: type[ResultType],
+) -> Any:
+    print(f"Return result: {result}")
+    if isinstance(result, ResultResponse) and isinstance(result.result, expected_type):
+        return result.result.value
+    raise ValueError(f"Expected {expected_type.__name__} result got: {result}")
+
+
+def return_ack(result: Union[AckResponse, ResultResponse, ErrResponse]) -> None:
+    print(f"Return ack: {result}")
+    if isinstance(result, AckResponse):
+        return
+    if isinstance(result, ErrResponse):
+        raise ValueError(result.message)
+    raise ValueError(f"Unexpected response type: {type(result).__name__}")
+
+
+async def get_webview_bin(options: WebViewOptions) -> str:
+    if "WEBVIEW_BIN" in os.environ:
+        return os.environ["WEBVIEW_BIN"]
+
+    flags = "-devtools" if cast(bool, getattr(options, "devtools", False)) else ""
+    if (
+        not flags
+        and cast(bool, getattr(options, "transparent", False))
+        and platform.system() == "Darwin"
+    ):
+        flags = "-transparent"
+
+    cache_dir = get_cache_dir()
+    file_name = f"deno-webview-{BIN_VERSION}{flags}"
+    if platform.system() == "Windows":
+        file_name += ".exe"
+    file_path = cache_dir / file_name
+
+    if file_path.exists():
+        return str(file_path)
+
+    url = f"https://github.com/zephraph/webview/releases/download/webview-v{BIN_VERSION}/deno-webview"
+    if platform.system() == "Darwin":
+        url += "-mac"
+        if platform.machine() == "arm64":
+            url += "-arm64"
+    elif platform.system() == "Linux":
+        url += "-linux"
+    elif platform.system() == "Windows":
+        url += "-windows.exe"
+    else:
+        raise ValueError("Unsupported OS")
+
+    url += flags
```

Comment:
This looks like unreachable code due to the `else`?

Reply:
It's not. The other parts of the `if` don't return, so the `else` is only triggered for unsupported platforms.

Reply:
Ah! You're right. I missed that.